content/rust-lld-on-stable-1.90.0/index.md

```diff
@@ -0,0 +1,65 @@
++++
+path = "2025/08/29/rust-lld-on-1.90.0-stable"
+title = "Faster linking times with 1.90.0 stable on Linux using `rust-lld`"
+authors = ["Rémy Rakic"]
+
+[extra]
+team = "the compiler performance working group"
+team_url = "https://www.rust-lang.org/governance/teams/compiler#team-wg-compiler-performance"
++++
+
+TL;DR: rustc will use `rust-lld` by default on `x86_64-unknown-linux-gnu` on stable, starting in 1.90.0, to significantly reduce linking times. Test it out on beta now, and please report issues if you encounter any.
+
+#### Some context
+
+Linking time is often a big part of compilation time. When rustc needs to build a binary or a shared library, it will usually call the default linker installed on the system to do that (this can be changed on the command-line or by the target for which the code is compiled).
+
+The linkers do an important job, with concerns about stability, backwards-compatibility and so on. For these and other reasons, on the most popular operating systems they usually are older programs, designed when computers only had a single core. So, they usually tend to be slow on a modern machine. For example, when building ripgrep 13 in debug mode on Linux, roughly half of the time is actually spent in the linker.
+
+There are different linkers, however, and the usual advice to improve linking times is to use one of these newer and faster linkers, like LLVM's [`lld`](https://lld.llvm.org/) or Rui Ueyama's [`mold`](https://github.com/rui314/mold).
+
+Some of Rust's wasm and aarch64 targets already use `lld` by default. When using rustup, rustc ships with a version of `lld` for this purpose. When CI builds LLVM to use in the compiler, it also builds the linker and packages it. It's referred to as `rust-lld` to avoid colliding with any `lld` already installed on the user's machine.
+
+Since improvements to linking times are substantial, it would be a good default to use in the most popular targets. This has been discussed for a long time, for example in issues [#39915](https://github.com/rust-lang/rust/issues/39915) and [#71515](https://github.com/rust-lang/rust/issues/71515).
+
+To expand our testing, we have enabled rustc to uses `rust-lld` by default on nightly, [in May 2024](https://blog.rust-lang.org/2024/05/17/enabling-rust-lld-on-linux.html). No major issues have been reported since then.
```

Comment:
```suggestion
To expand our testing, we have enabled rustc to use `rust-lld` by default on nightly, [in May 2024](https://blog.rust-lang.org/2024/05/17/enabling-rust-lld-on-linux.html). No major issues have been reported since then.
```